src/components/ArticlePage.jsx

```diff
@@ -4,25 +4,57 @@ import { useState, useEffect } from 'react';
 import { fetchArticleById } from '../../api';
 import { formatDate } from '../utils/utils';
 import Comments from './Comments';
+import { patchArticleVotes } from '../../api';
 
 function ArticlePage() {
     const { articleId } = useParams();
-    const [article, setArticle] = useState({});
+    const [article, setArticle] = useState(null);
+    const [votes, setVotes] = useState(0);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
 
     useEffect(() => {
+      setLoading(true)
         fetchArticleById(articleId).then((article) => {
             setArticle(article);
-        });
+            setVotes(article.votes);
+            setError(null)
+            setLoading(false)
+        })
+        .catch((err) => {
+          setError(err.message)
+          setLoading(false)
+        })
     }, [articleId]);
 
+    if (loading) {
+      return <p>Loading article...</p>
+    } 
+    
+    if (error) {
+      return <p>{error}</p>
+    }
+
+
+    const handleVote = (increment) => {
+      setVotes((preVotes) => preVotes + increment)
+      patchArticleVotes(article.article_id, increment)
+      .then((updatedArticle) => {
+        setVotes(updatedArticle.votes)
```

Comment:
in this task, we need optimistic rendering - when the internet is slow, the user still can see the change instantly without needing to wait for the data back from the api. You've updated votes twice here (line 40 and line 43) which is a bit unnecessary. Could only update it on line 40 and display `article.votes + votes` on line 55